components/Tabs/Tabs.js

```diff
@@ -1,38 +1,58 @@
 class TabLink {
   constructor(element){
     // assign this.element to the element reference
-    this.element;
+    this.element = element;
     // Get the tab data attribute and save the value here
-    this.tabData = ; 
+    this.tabData = this.element.dataset.tab; 
+
     // Find all elements with the .card class in index.html that correspond to the tab data attribute
     // If the data is 'all' then select all cards regardless of their data attribute
-    if(){
-      this.cards = ;
+
+  
+    if(this.cards){
+      this.cards = document.querySelector('.card');
     } else {
-      this.cards = ;
+      this.cards = document.querySelectorAll(`.card[data-tab = "${this.tabData}"]`);
     }
 
+    // if(this.cards){
+    //   this.cards = document.querySelector(`.card[data-tab = "${this.tabData}"]`);
+    // } else {
+    //   this.cards = document.querySelectorAll(`.card[data-tab = "${this.tabData}"]`);
+    // }
+
     // Map over the cards array and convert each card element into a new instance of the TabCard class. 
     // Pass in the card object to the TabCard class.
-    this.cards = Array.from(this.cards).map();
+    this.cards = Array.from(this.cards).map(card => new TabCard(card));
+
     // Add a click event that invokes selectTab
-    this.element.addEventListener();
+    this.element.addEventListener('click', ()=> {this.selectTab()
+  });
   }
 
   selectTab(){
 
     // Select all elements with the .tab class on them
-    const tabs = document.querySelectorAll();
+    const tabs = document.querySelectorAll('.tab');
+
     // Iterate through the NodeList removing the .active-tab class from each element
-    tabs.forEach()
+
+    tabs.forEach(function(item){
```

Comment:
Good. Get used to ES6 syntax, however. An arrow function here instead of the ES5 `function` keyword would be great here.